src/main/proto/com/google/api/codegen/config.proto

```diff
@@ -333,13 +333,28 @@ message SampleValueSet {
   //   several placeholders.
   repeated string parameters = 4;
 
-  repeated string sample_args = 5;
+  // Attributes governing how to init request object.
+  // Each attribute must have unique, non-overlapping, path.
+  repeated SampleInitAttribute attributes = 5;
 
   // How to process a successful response from the client library
   // function being exemplified.
   repeated OutputSpec on_success = 6;
 }
 
+// `SampleInitAttribute` controls how a sample initializes the request object.
+message SampleInitAttribute {
+  // Required. The part of the request object this attribute is talking about.
+  string path = 1;
+
+  // If a sample, the call to client lib method is itself in a function.
+  // If `function_param` is true, the part of the request object specified by `path`
+  // should be accepted as a parameter to the sample function.
+  //
+  // TODO: Allow configuring the param name?
```

Comment:
@vchudnov-g How important is this? I'm not sure how to do this yet, but I can prioritize investigating.

Reply:
Let's loop in customers next week about this. I can see it becoming important.

Reply:
Yeah, let's defer this for after this PR, but we'll need a user-friendly solution before releasing. We can discuss which of the two makes more sense when.